internal/sorting/sorting_test.go

```diff
@@ -0,0 +1,471 @@
+			}
+		})
+	}
+}
+
+func TestNaturalSort(t *testing.T) {
+	t.Parallel()
+	type testCase struct {
+		testing       string
+		lines         []bed.Line
+		expectedLines []bed.Line
+	}
+	testCases := []testCase{
+		{
+			testing: "chr sort",
+			lines:   deepCopyBedLines(testChrSort),
+			expectedLines: []bed.Line{
+				{
+					Chr: "chr1", Start: 8, Stop: 9,
+					Full: []string{"1", "8", "9"},
+				},
+				{
+					Chr: "chr2", Start: 8, Stop: 9,
+					Full: []string{"1", "8", "9"},
+				},
+				{
+					Chr: "chr10", Start: 8, Stop: 9,
+					Full: []string{"1", "8", "9"},
+				},
+				{
+					Chr: "chrX", Start: 8, Stop: 9,
+					Full: []string{"1", "8", "9"},
+				},
+				{
+					Chr: "GL000209.1", Start: 8, Stop: 9,
+					Full: []string{"1", "8", "9"},
+				},
+				{
+					Chr: "HG385_PATCH", Start: 8, Stop: 9,
+					Full: []string{"1", "8", "9"},
+				},
+			},
+		},
+		{
+			testing: "full sort",
+			lines:   deepCopyBedLines(testFullSort),
+			expectedLines: []bed.Line{
+				{
+					Chr: "1", Start: 8, Stop: 9,
+					Strand: "-1", Feat: "B",
+					Full: []string{"1", "8", "9", "-1", "B"},
+				},
+				{
+					Chr: "1", Start: 10, Stop: 11,
+					Strand: "-1", Feat: "A",
+					Full: []string{"1", "10", "11", "-1", "A"},
+				},
+				{
+					Chr: "1", Start: 10, Stop: 11,
+					Strand: "-1", Feat: "B",
+					Full: []string{"1", "10", "11", "-1", "B"},
+				},
+				{
+					Chr: "1", Start: 10, Stop: 11,
+					Strand: "1", Feat: "A",
+					Full: []string{"1", "10", "11", "1", "A"},
+				},
+				{
+					Chr: "1", Start: 12, Stop: 13,
+					Strand: "1", Feat: "A",
+					Full: []string{"1", "12", "13", "1", "A"},
+				},
+				{
+					Chr: "2", Start: 12, Stop: 13,
+					Strand: "1", Feat: "C",
+					Full: []string{"2", "12", "13", "1", "C"},
+				},
+				{
+					Chr: "10", Start: 12, Stop: 13,
+					Strand: "1", Feat: "D",
+					Full: []string{"10", "12", "13", "1", "D"},
+				},
+				{
+					Chr: "GL000209.1", Start: 10, Stop: 11,
+					Strand: "1", Feat: "A",
+					Full: []string{"GL000209.1", "10", "11", "1", "A"},
+				},
+				{
+					Chr: "HG385_PATCH", Start: 10, Stop: 11,
+					Strand: "1", Feat: "A",
+					Full: []string{"X", "10", "11", "1", "A"},
+				},
+				{
+					Chr: "X", Start: 10, Stop: 11,
+					Strand: "1", Feat: "A",
+					Full: []string{"X", "10", "11", "1", "A"},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.testing, func(t *testing.T) {
+			t.Parallel()
+			receivedLines := naturalSort(tc.lines)
+			if diff := deep.Equal(tc.expectedLines, receivedLines); diff != nil {
+				t.Error("expected VS received lines", diff)
+			}
+		})
+	}
+}
+
+func TestMergeSort(t *testing.T) {
+	t.Parallel()
+	type testCase struct {
+		testing       string
+		lines         []bed.Line
+		expectedLines []bed.Line
+	}
+	testCases := []testCase{
+		{
+			testing: "chr sort",
+			lines:   deepCopyBedLines(testChrSort),
+			expectedLines: []bed.Line{
+				{
+					Chr: "GL000209.1", Start: 8, Stop: 9,
```

Comment:
maybe I did not understand the function correctly but I expected the chromosomes to be in the correct order with no feature/the same feature present in all lines?

Reply:
This sorting is meant to be used for merging functionality that is to come, so we need to sort in a different way that will probably not make sense for users. The order of the chromosomes in this case therefore does not make sense at all as we sort on feature first then strand and lastly chromosome. Also because we do not lower the strings for the this sorting the GL000209.1 comes out on top as `natural.Less` seems to classify capital letters as lower than lowercase letters.

Reply:
yes, then it makes sense, thank you 😺